metric-publishers/emf-metric-publisher/src/main/java/software/amazon/awssdk/metrics/publishers/emf/internal/MetricEmfConverter.java

```diff
@@ -0,0 +1,339 @@
+
+        jsonWriter.writeEndObject(); // End root object
+
+        return new String(jsonWriter.getBytes(), StandardCharsets.UTF_8);
+
+    }
+
+    private void writeAwsObject(JsonWriter jsonWriter, Map<String, Class<?>> metricNames) {
+        jsonWriter.writeFieldName("_aws");
+        jsonWriter.writeStartObject();
+
+
+        jsonWriter.writeFieldName("Timestamp");
+        //Unit Test
+        jsonWriter.writeValue(clock.instant());
+
+
+        jsonWriter.writeFieldName("LogGroupName");
+        jsonWriter.writeValue(config.getLogGroupName());
+
+        writeCloudWatchMetricsArray(jsonWriter, metricNames);
+        jsonWriter.writeEndObject();
+    }
+
+    private void writeCloudWatchMetricsArray(JsonWriter jsonWriter, Map<String, Class<?>> metricNames) {
+        jsonWriter.writeFieldName("CloudWatchMetrics");
+        jsonWriter.writeStartArray();
+
+        writeCloudWatchMetricsObjects(jsonWriter, metricNames);
+        jsonWriter.writeEndArray();
+    }
+
+    private void writeCloudWatchMetricsObjects(JsonWriter jsonWriter,  Map<String, Class<?>> metricNames) {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeFieldName("Namespace");
+        jsonWriter.writeValue(config.getNamespace());
+
+        writeDimensionSetArray(jsonWriter);
+
+        writeMetricDefinitionArray(jsonWriter, metricNames);
+        jsonWriter.writeEndObject();
+    }
+
+    private void writeDimensionSetArray(JsonWriter jsonWriter) {
+        jsonWriter.writeFieldName("Dimensions");
+        jsonWriter.writeStartArray();
+        jsonWriter.writeStartArray();
+        for (String dimension : realDimensionStrings) {
+            jsonWriter.writeValue(dimension);
+        }
+        jsonWriter.writeEndArray();
+        jsonWriter.writeEndArray();
+    }
+
+    private void writeMetricDefinitionArray(JsonWriter jsonWriter,  Map<String, Class<?>> metricNames) {
+        jsonWriter.writeFieldName("Metrics");
+        jsonWriter.writeStartArray();
+
+
+        // Write metric definitions
+        metricNames.forEach((name, type) -> {
+            jsonWriter.writeStartObject();
+            jsonWriter.writeFieldName("Name");
+            jsonWriter.writeValue(name);
+
+            String unit = getMetricUnit(type);
+            if (unit != null) {
+                jsonWriter.writeFieldName("Unit");
+                jsonWriter.writeValue(unit);
+            }
+
+            jsonWriter.writeEndObject();
+        });
+
+        jsonWriter.writeEndArray();
+    }
+
+
+    private void writeMetricValues(JsonWriter jsonWriter, Map<String, List<MetricRecord<?>>> metrics) {
+        for (Map.Entry<String, List<MetricRecord<?>>> entry : metrics.entrySet()) {
+            String metricName = entry.getKey();
+            List<MetricRecord<?>> records = entry.getValue();
+
+            if (isDimension(metricName)) {
+                jsonWriter.writeFieldName(metricName);
+                jsonWriter.writeValue((String) records.get(0).value());
```

Comment:
I understand dimension metric could only have one element, but for the sake of defensive coding, let's add validation here.